src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            List<Bulldozer> bulldozers = bulldozerProducer.get();
+            return (List<T>) bulldozers;
+        } else if (type == Excavator.class) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            List<Excavator> excavators = excavatorProducer.get();
+            return (List<T>) excavators;
+        } else if (type == Truck.class) {
+            TruckProducer truckProducer = new TruckProducer();
+            List<Truck> trucks = truckProducer.get();
+            return (List<T>) trucks;
+        } else {
+            List machines = new ArrayList<>();
+            return (List<T>) machines;
+        }
```

Comment:
you can producers class-level fields to avoid creating them each time this method is invoked